### ISSUE TITLE
Request - Commands for Osara

### ISSUE THREAD

**Issue by ElCrackLoko** [jcsteh/osara#549]
Hello.
I know that you can customize an action in Reaper from the keymap, but I would like to make a request in relation to Osara, if possible of course.
First, if it would be possible to return the Escape key to exit the MIDI editor more quickly, (As in the old KeyMap) since pressing Alt + F4 is not the fastest way to close the editor, although it can be left to those who like it that way. Or perhaps, add the setting to enable / disable the escape key to exit the MIDI editor. I mean, if you can, it would be excellent ...
Second, this request is on a personal level. Those of us who do Spots with Reaper, (or at least I do) sometimes have to pan a shot to different positions. Example, one on the right, one on the left, and sometimes these must be so close that making an automation does not seem the solution, so it is best to press Alt + P and pan take by shot. I've done that job with up to 20 takes, and I feel like adding that command (Or if it exists please tell me) we could speed up our work environment.
I mean, we have a command to raise and lower the volume of the Items, also of the takes, so why not also add one to pan them?
Thanks.

**Comment by ScottChesworth** [comment]
> First, if it would be possible to return the Escape key to exit the MIDI editor more quickly, (As in the old KeyMap) since pressing Alt + F4 is not the fastest way to close the editor.
> 
This has been the least popular key map change ever. I can revert it, just need to figure out where to put removal of time selection while inside the MIDI Editor. Any ideas?

> we have a command to raise and lower the volume of the Items, also of the takes, so why not also add one to pan them?
> 
Plenty of actions are already available via SWS. Here's a few options:
1. You could select all takes that you want to pan to a matching position, then run "SWS/S&M: Pan active takes of selected items to 25% left". Other variants are available to pan 50%, 75% and 100% in either direction.
2. In situations where you want to pan multiple takes to a position that doesn't match any of the options above, try "Xenakios/SWS: Set volume and pan of selected takes", currently assigned to Alt+S.

None of those actions provide spoken feedback yet. If you need that, I can try to get it going.

**Comment by KeithRempel** [comment]
Not sure how much mileage this suggestion will have, given the unintuitive nature of my choice, but I have remapped the clear time selection to the key immediately below the escape key to the left of number 1. This did overwrite another assignment, but for something I do not use.

That particular key is maybe not the answer. But Getting clear time selection away from escape has saved me from inadvertently erasing time selection when whacking escape to get out of menus Etc, which is something that used to happen to me several times per week.

**Comment by mattymc2** [comment]
Have to agree on this point. I do bash the escape key a lot and cancel time selections which can be frustrating. Could remove times selection be on shift escape or something like that?

**Comment by ElCrackLoko** [comment]
Hello everybody.
@ScottChesworth
The truth was I was not aware that you could use the time selection in the MIDI editor, I offer an apology for opening a nonsensical thread without first finding out or fiddling there jajaja. I think the time selection as it is is fine for me. For the comrades who comment that they eliminate the time selection when pressing Escape, it is a bad habit that we must remove, since the only thing that was closed with Escape before was only the MIDI editor and the virtual keyboard, now, it is only the latter. You can always use the CTRL + Z command to scroll the changes, such as removing the time selection.
Regarding the action of SWS, I knew that it could be assigned, but I did not know the command Alt + S to be able to quickly control the panorama and even the volume and, incidentally, in a personalized way. Again, I'm the one to blame for not finding out jajajajaja. I could have done it as a query instead of as a request jajajaja.
We can close the thread if it seems so, since my query is already solved.

**Comment by ScottChesworth** [comment]
> I offer an apology for opening a nonsensical thread without first finding out or fiddling
> 
No apology needed dude. It happens to us all sometimes. :)

Keeping this open for a bit anyway to debate the time selection assignment, reverting that behaviour has been requested via other channels many times. I'll close the thread once that's settled.

**Comment by ScottChesworth** [comment]
> Could remove times selection be on shift escape or something like that?
> 
Shift+Escape is already being used, "Unselect all tracks/items/envelope points". What else ya got? Seems like between Windows and Mac, all other modifiers in combination with Escape do OS things.

**Comment by mattymc2** [comment]
Alt shift esc? Not really intuitive though. Probably best to just leaving it where it is

**Comment by KeithRempel** [comment]
Interesting idea. opt shift escape is free on mac. Could move the unselect all tracks items envelope points to alt shift escape and put time selection on shift escape. Feels a bit intuitive having the remove time selection associated with shift, and adding in the alt to get rid of tracks and items.

**Comment by Lo-lo78** [comment]
Hi guys, it's a very popular command, ctrl + w. For me it is perfect, 
many applications use ctrl + w to close a window or a tab. It is 
standard. Regards!

**Comment by ScottChesworth** [comment]
> Could move the unselect all tracks items envelope points to alt shift escape and put time selection on shift escape.
> 
No can do gov. Alt+Escape is a switching command in Windows, and adding the Shift makes no difference to whether it's assignable in Reaper. As soon as you add that Alt key, all bets are off with Escape, doesn't matter how many other modifiers you throw in.

**Comment by ScottChesworth** [comment]
> Many applications use ctrl + w to close a window or a tab. It is standard.
> 
Yup, that standardization is even stronger over on Mac with Command+W which we already have mapped. Ditto strong standardization with Control+F4 and Alt+F4 in Windows, both of those work too.
All that said, no matter how much I keep reassuring people they'll adapt to the change, it ain't happening. :)

**Comment by RDMurray** [comment]
Escape also closes every other window in Reaper.  Apologies, it was me who first suggested remapping escape in the MIDI editor on the keymap group. :)  However, I bet that changing clear time selection to a different key would also be unpopular.  That said, that would probably be the best option, given that we can find an appropriate key.
